projects/ngneat/reactive-forms/src/lib/formGroup.spec.ts

```diff
@@ -285,4 +295,44 @@ describe('FormGroup', () => {
     control.patchValue({ name: 'Test' });
     expect(spy).toHaveBeenCalledWith({ invalidName: true });
   });
+
+  describe('.persist()', () => {
+    const person: Person = { name: 'ewan', phone: { num: 5550153, prefix: 288 }, skills: ['acting', 'motorcycle'] };
+
+    it('should persist', fakeAsync(() => {
+      const control = createGroup();
+      const debounceTime = 50;
+      const persistManager = {
+        getValue: jest.fn(),
+        setValue: jest.fn()
+      };
+      control.persist('key', { debounceTime, manager: persistManager }).subscribe();
+      control.getControl('name').setValue('ewan');
+      tick(debounceTime);
+      expect(persistManager.setValue).toHaveBeenCalledWith('key', control.value);
+    }));
+
+    it.each([[person], [Promise.resolve(person)], [of(person)]])(
```

Comment:
Can you please add specs for async storage?

Reply:
do you mean for `setValue`?

Reply:
I mean when the manager is async. (both getValue and setValue)

Reply:
Changed the tests so that the mock gets/sets value after a delay